index.js

```diff
@@ -44,17 +45,41 @@ var initConfig = function() {
     config.appPathRelativeToRepository =
       process.env.GCLOUD_DEBUG_REPO_APP_PATH;
   }
+  if (process.env.hasOwnProperty('GAE_MODULE_NAME')) {
+    config.serviceName = process.env.GAE_MODULE_NAME;
+  }
+  if (process.env.hasOwnProperty('GAE_MODULE_VERSION')) {
+    config.serviceVersion = process.env.GAE_MODULE_VERSION;
+  }
   return config;
 };
 
-// exports is populated by the agent
-module.exports = {};
-var config = initConfig();
-
-var log = logger.create(config.logLevel, '@google/cloud-debug');
+var hasStarted = false;
+var log;
+function start(config_) {
+  if (hasStarted) {
+    return log.error('The cloud-debug agent has already been started.');
+  }
 
-if (config.enabled) {
-  var debuglet = new Debuglet(config, log);
-  debuglet.start();
-  module.exports.private_ = debuglet;
+  var config = initConfig(config_);
+  log = logger.create(config.logLevel, '@google/cloud-debug');
+  if (config.enabled) {
+    var debuglet = new Debuglet(config, log);
+    debuglet.start();
+    module.exports.private_ = debuglet;
+    hasStarted = true;
+  }
 }
+
+// exports is populated by the agent
+module.exports = {
+  start: start
+};
+
+setTimeout(function() {
+  if (!hasStarted){
+    start();
+    log.error('The cloud-debug agent has been automatically started.  ' +
+      'This action will be deprecated in the future.');
+  }
+}, 5*1000);
```

Comment:
Could we add a test to ensure that this correctly starts the agent and does not interfere with an already started agent?